extensions/wikia/PageHeader/Counter.class.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Wikia\PageHeader;
+
+use Forum;
+use MediaQueryService;
+use RequestContext;
+use WikiaApp;
+
+class Counter {
+	public function __construct( WikiaApp $app ) {
+		$title = RequestContext::getMain()->getTitle();
+
+		if ( $title->isSpecial( 'Videos' ) ) {
+			$this->message = $this->getMessageForSpecialVideos();
+		} else if ( $title->isSpecial( 'Images' ) ) {
+			$this->message = $this->getMessageForSpecialImages();
+		} else if ( defined( 'NS_BLOG_LISTING' ) && $title->inNamespace( NS_BLOG_LISTING ) ) {
+			$this->message = $this->getMessageForBlogListing( $app );
+		} else if ( $title->isSpecial( 'Forum' ) ) {
+			$this->message = $this->getMessageForForum();
+		}
+	}
+
+	private function getMessageForSpecialVideos() {
+		$mediaService = ( new MediaQueryService );
+		$count = $mediaService->getTotalVideos();
+
+		return wfMessage( 'page-header-counter-videos', $count )->parse();
+	}
+
+	private function getMessageForSpecialImages() {
+		$mediaService = ( new MediaQueryService );
+		$count = $mediaService->getTotalImages();
+
+		return wfMessage( 'page-header-counter-images', $count )->parse();
+	}
+
+	private function getMessageForBlogListing( WikiaApp $app ) {
+		$count = $app->wg->Parser->getOutput()->getProperty( 'blogPostCount' );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Do not use the global wgParser Parser instance. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWgParserUsage)